src/sql/plot.py

```diff
@@ -21,15 +21,24 @@
 
 def _summary_stats(con, table, column, with_=None):
     """Compute percentiles and mean for boxplot"""
+    # Calculating all percentiles in 1 pass through the data then pulling apart the list
+    # (This improves performance by ~40%)
     template = Template(
         """
+WITH stats as (
+    SELECT
+    percentile_disc([0.25, 0.50, 0.75]) WITHIN GROUP (ORDER BY "{{column}}") AS percentiles,
+    AVG("{{column}}") AS mean,
+    COUNT(*) AS N
+    FROM "{{table}}"
+)
 SELECT
-percentile_disc(0.25) WITHIN GROUP (ORDER BY "{{column}}") AS q1,
-percentile_disc(0.50) WITHIN GROUP (ORDER BY "{{column}}") AS med,
-percentile_disc(0.75) WITHIN GROUP (ORDER BY "{{column}}") AS q3,
-AVG("{{column}}") AS mean,
-COUNT(*) AS N
-FROM "{{table}}"
+percentiles[1] as q1,
+percentiles[2] as median,
+percentiles[3] as q3,
+mean,
+N
+FROM stats
```

Comment:
we currently do not support CTEs in this template sine it's later wrapped into one. the CI is failing because the final query ends up in two nested WITH statements.

@Alex-Monahan: would re-writing this as a subquery have any impact?

````
SELECT * FROM (SELET * FROM ...)
````

Reply:
@tonykploomber  Let's make the tests pass first; we need to re-write Alex's change as a subquery since he wrote a CTE, and we don't support that now.

Then, we need to check if `percentile_disc([0.25, 0.50, 0.75])` is supported by other major database vendors, if not, then we might have to write two versions of this query, a generic one and one optimized for DuckDB.

Finally, we can write some benchmarks to see if Alex's suggested change improves performance when running the code as a subquery.

Reply:
I changed the template to:
```
SELECT
percentiles[1] as q1,
percentiles[2] as med,
percentiles[3] as q3,
mean,
N
FROM (
    SELECT
    percentile_disc([(0.25), (0.50), (0.75)]) WITHIN GROUP (ORDER BY "{{column}}") AS percentiles,
    AVG("{{column}}") AS mean,
    COUNT(*) AS N
    FROM "{{table}}"
)
```

Somehow it's still fail, need to check more on this

Reply:
Since I don't have permission on this branch...

I created another one as https://github.com/ploomber/jupysql/pull/152

We may move the discussion over there

Reply:
sure, I'll close this one